src/pages/gen2/build-a-backend/data/customize-authz/grant-lambda-function-access-to-api/index.mdx

```diff
@@ -11,31 +12,143 @@ export function getStaticProps(context) {
-
-To grant an external AWS resource or an IAM role access to the Amplify Data API, you need to explicitly list the IAM roles by adding them to `allowListedRoleNames` property.
+Function access to `defineData` can be configured using an authorization rule on the schema object.
 
 ```ts title="amplify/data/resource.ts"
-// amplify/data/resource.ts
-import { a, defineData, type ClientSchema } from "@aws-amplify/backend";
-
-const schema = a.schema({
-  Todo: a.model({
-    name: a.string(),
-    description: a.string(),
-  }),
+import {
+  a,
+  defineData,
+  defineFunction,
+  type ClientSchema
+} from '@aws-amplify/backend';
+
+const functionWithDataAccess = defineFunction({
+  entry: '../functions/data-access.ts'
 });
 
+const schema = a
+  .schema({
+    Todo: a.model({
+      name: a.string(),
+      description: a.string()
+    })
+  })
+  // highlight-next-line
+  .authorization([a.allow.resource(functionWithDataAccess)]);
+
 export type Schema = ClientSchema<typeof schema>;
 
 export const data = defineData({
-  schema,
-  authorizationModes: {
-    // Pass in the Lambda Execution Role names to grant full read/write access to the API
-    // if their IAM policies permit it.
-    allowListedRoleNames: ["lambdaRole"],
-  },
+  schema
 });
 ```
 
-These "Allow-listed Roles" have special access privileges that are scoped based on their [IAM policy](https://docs.aws.amazon.com/appsync/latest/devguide/security-authz.html#aws-iam-authorization) instead of any particular `.authorization()` rule.
+The object returned from `defineFunction` can be passed directly to to `a.allow.resource()` in the schema authorization rules. This will grant the function the ability to execute Query, Mutation, and Subscription operations against the GraphQL API. Use the `.to()` method to narrow down access to one or more operations.
+
+```ts
+const schema = a
+  .schema({
+    Todo: a.model({
+      name: a.string(),
+      description: a.string()
+    })
+  })
+  // highlight-next-line
+  .authorization([
+    a.allow.resource(functionWithDataAccess).to(['query', 'listen'])
+  ]); // allow query and subscription operations but not mutations
+```
+
+When configuring function access, the function will be provided the API endpoint as an environment variable named `<defineDataName>_GRAPHQL_ENDPOINT`. The default name is `amplifyData_GRAPHQL_ENDPOINT` unless you have specified a different name in `defineData`.
+
+<Callout info>
+
+Function access can only be configured on the schema object. It cannot be configured on individual models or fields.
+
+</Callout>
+
+## Access the API using `aws-amplify`
+
+<Callout warning>
+
+**Experimental:** Configuring the `aws-amplify` data client within a Lambda function is under development. There are rough edges to the experience that we are actively working on.
```

Comment:
Can we use a similar callout as the custom resource pages?
https://docs.amplify.aws/gen2/build-a-backend/add-aws-services/rest-api/
![image](https://github.com/aws-amplify/docs/assets/5033303/dbb08f52-b6e0-42bc-b497-d6fe85e3be84)

Reply:
updated